src/api/spec/models/workflow_spec.rb

```diff
@@ -43,10 +43,30 @@
       end
       let!(:target_project) { create(:project, name: 'test-target-project:openSUSE:open-build-service:PR-4', maintainer: user) }
 
-      before { login user }
-
-      it 'removes the target project' do
-        expect { subject.call }.to change(Project, :count).from(2).to(1)
+      context 'we are dealing with a branch package step' do
+        before { login user }
+
+        it 'removes the target project' do
+          expect { subject.call }.to change(Project, :count).from(2).to(1)
```

Comment:
Would it be possible to also check that `EventSubscription` instances are deleted?